var/spack/repos/builtin/packages/upcxx/package.py

```diff
@@ -76,7 +102,72 @@ def setup_dependent_build_environment(self, env, dependent_spec):
             env.set('UPCXX_NETWORK', 'aries')
 
     def install(self, spec, prefix):
-        env['CC'] = self.compiler.cc
-        env['CXX'] = self.compiler.cxx
-        installsh = Executable("./install")
-        installsh(prefix)
+        # UPC++ follows autoconf naming convention for LDLIBS, which is 'LIBS'
+        if (env.get('LDLIBS')):
+            env['LIBS'] = env['LDLIBS']
+
+        if spec.version <= Version('2019.9.0'):
```

Comment:
These seem to be pretty different paths -- if you want you can have two install methods like this -- might be cleaner:

```python
def install(self, spec, prefix):
    # default logic

@when("@:2019.9.0")
def install(self, spec, prefix):
    # logic for 2019.9.0 and before
```